### ISSUE TITLE
Destructuring binding assignment is not supported

### ISSUE THREAD

**Issue by dangduomg** [rakudo/rakudo#5194]
## The Problem
Destructuring binding declaration is possible, but the corresponding assignment is not possible as exemplified by this code here:
```
my ($f, $f-next) := (0, 1);

($f, $f-next) := ($f-next, $f + $f-next);

say $f, $f-next;
```

## Expected Behavior
```
12
```

## Actual Behavior
```
===SORRY!=== Error while compiling C:\Users\jack9\Documents\raku/thonk.raku
Cannot use bind operator with this left-hand side
at C:\Users\jack9\Documents\raku/thonk.raku:3
```
Notably, if I replace the problematic line with `:($f, $f-next) := ($f-next, $f + $f-next);`, it will work as expected

## Environment
* Operating system: Windows 10
* Compiler version (`perl6 -v` or `raku -v`): v6.d

**Comment by lizmat** [comment]
Feels like a duplicate of https://github.com/rakudo/rakudo/issues/4522

**Comment by dangduomg** [comment]
really?

**Comment by jnthn** [comment]
I think it's distinct from #4522. The issue in #4522 is relatively clear: we should compile it as if it was a bunch of attribute declarations. Since we parse a signature after the `has`, there's no issues with doing that. It's a bit painful in the current compiler frontend, and probably quite tractable to do in the RakuAST-based one.

By contrast, this issue is a feature request, and probably an awkward one. Destructuring is always performed via signature binding. When one writes:

```
my ($f, $f-next) := (0, 1);
```

Then we parse a signature after the `my`; it's as if there's a `:` before the `(`. By contrast, when we parse:

```
($f, $f-next) := ($f-next, $f + $f-next);
```

Then it's the standard expression parser at work, meaning it parses the parens just as usual grouping parens, and the `,` as the usual comma operator.  Thus we have a `List`, not a `Signature`.

It's true that in the case of:

```
my ($head, @rest) = foo();
```

The parsed `Signature` is then instead code-gen'd to produce a `List` of the two declared variables. One might ask if we could promote the `List` into a `Signature` in the compiler somehow, and it could be made to deal with the case mentioned in this issue. But the moment you actually try and do some interesting destructuring, taking a `Pair` apart with:

```
($pair (:$key, :$value)) := foo();
```

Then you'll have a two terms in a row syntax error. Even if one accepts it's going to be limited, then there'l be all the questions like what to do about:

```
((:$key, :$value)) := foo();
```

Since the characters are the same as something that would parse into a signature successfully, but it will have been interpreted very differently before the point we see the `:=` operator; while the signature AST can easily have a list construction AST extracted from it, that's not true the other way around.

In short, it's probably best to just require the `:(...)` form we used. If one wants consistency, one can always write the `:` in the `my` case too:

```
my :($foo, $bar) := (1, 2)
```
